src/DynamoCore/Interfaces/IWorkspaceModel.cs

```diff
@@ -21,5 +21,11 @@ public interface IWorkspaceModel
         event Action NodesCleared;
         event Action<ConnectorModel> ConnectorAdded;
         event Action<ConnectorModel> ConnectorDeleted;
+
+        /// <summary>
+        /// Implement to record node modification for undo/redo
+        /// </summary>
+        /// <param name="nodeModel"></param>
+        void RecordModelForModification(ModelBase nodeModel);
```

Comment:
Maybe I'm mistaken, but why would we need to record models for undo at all? It seems like code that we should keep away from external clients. Shouldn't that be handled by the CommandExecutive?